systems/subscriber/subscriber-registry/pkg/db/subscriber_repo.go

```diff
@@ -0,0 +1,84 @@
+package db
+
+import (
+	"github.com/gofrs/uuid"
+	"github.com/ukama/ukama/systems/common/sql"
+	"gorm.io/gorm"
+)
+
+type SubscriberRepo interface {
+	Add(subscriber *Subscriber) error
+	Get(subscriberID uuid.UUID) (*Subscriber, error)
+	Delete(subscriberID uuid.UUID) error
+	Update(subscriberID uuid.UUID, sub Subscriber) (*Subscriber, error)
+	GetByNetwork(networkID uuid.UUID) ([]Subscriber, error)
+	ListSubscribers() ([]Subscriber, error)
+}
+
+type subscriberRepo struct {
+	Db sql.Db
+}
+
+func NewSubscriberRepo(db sql.Db) *subscriberRepo {
+	return &subscriberRepo{
+		Db: db,
+	}
+}
+
+func (s *subscriberRepo) Add(subscriber *Subscriber) error {
+	db := s.Db.GetGormDb()
+	return db.Create(subscriber).Error
+}
+func (s *subscriberRepo) ListSubscribers() ([]Subscriber, error) {
+
+	var subscribers []Subscriber
+	result := s.Db.GetGormDb().Find(&subscribers)
+	if result.Error != nil {
+		return nil, result.Error
+	}
+	return subscribers, nil
+}
+
+func (s *subscriberRepo) Get(subscriberId uuid.UUID) (*Subscriber, error) {
+	var subscriber Subscriber
+	
+	err := s.Db.GetGormDb().Where("subscriber_id = ?", subscriberId).FirstOrCreate(&subscriber).Error
```

Comment:
Should the call to get a given subscriber, create a new record if that subscriber does not exist?

Reply:
I have changed it to call .`First` instead .